addons/payment_sips/models/payment.py

```diff
@@ -144,16 +127,12 @@ def _sips_form_get_tx_from_data(self, data):
         reference = data.get('transactionReference')
 
         if not reference:
-            custom = json.loads(data.pop('returnContext', False) or '{}')
-            reference = custom.get('reference')
+            return_context = json.loads(data.get('returnContext', '{}'))
+            reference = return_context.get('reference')
 
         payment_tx = self.search([('reference', '=', reference)])
-        if not payment_tx or len(payment_tx) > 1:
-            error_msg = _('Sips: received data for reference %s') % reference
-            if not payment_tx:
-                error_msg += _('; no order found')
-            else:
-                error_msg += _('; multiple order found')
```

Comment:
Was this intentionally dropped ?

Reply:
this can't ever happen, there is a unique constraint on the `reference` field in the payment base module

useless code is thus useless